.script/createComponent.ts

```diff
@@ -0,0 +1,138 @@
+import * as fs from 'node:fs/promises';
+import * as path from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { intro, outro, spinner, text } from '@clack/prompts';
+import { Command, Option } from 'clipanion';
+import { Cli } from 'clipanion';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+class CreateComponentCommand extends Command {
+  static paths = [['create']];
+  name = Option.String({ required: false });
+
+  private kebabToPascal(str: string): string {
+    return str
+      .split('-')
+      .map((part) => part.charAt(0).toUpperCase() + part.slice(1).toLowerCase())
+      .join('');
+  }
+
+  private replacePatterns(kebabCaseName: string, pascalCaseName: string) {
+    return {
+      Component: pascalCaseName,
+      'package-name': kebabCaseName,
+    } as const;
+  }
+
+  private excludePatterns = ['node_modules', '.git', '.DS_Store'];
+
+  async copyRecursive(
+    source: string,
+    target: string,
+    kebabCaseName: string,
+    pascalCaseName: string,
+  ) {
+    const currentFolder = path.basename(source);
+
+    if (this.excludePatterns.includes(currentFolder)) {
+      return;
+    }
+
+    const stats = await fs.stat(source);
+    const patterns = this.replacePatterns(kebabCaseName, pascalCaseName);
+
+    if (stats.isDirectory()) {
+      await fs.mkdir(target, { recursive: true });
+      const entries = await fs.readdir(source, { withFileTypes: true });
+
+      for (const entry of entries) {
+        if (!this.excludePatterns.includes(entry.name)) {
+          const sourcePath = path.join(source, entry.name);
+          const newName = entry.name.replace(/Component/g, pascalCaseName);
+          const targetPath = path.join(target, newName);
+          await this.copyRecursive(
+            sourcePath,
+            targetPath,
+            kebabCaseName,
+            pascalCaseName,
+          );
+        }
+      }
+    } else {
+      const content = await fs.readFile(source, 'utf-8');
+      const updatedContent = Object.entries(patterns).reduce(
+        (content, [search, replace]) =>
+          content.replace(new RegExp(search, 'g'), replace),
+        content,
+      );
+
+      await fs.writeFile(target, updatedContent);
+    }
+  }
+
+  async execute() {
+    const loading = spinner();
+
+    try {
+      intro('컴포넌트 생성기 🚀');
+
+      const kebabCaseName = (await text({
+        message: '생성할 컴포넌트 이름을 입력해주세요 (kebab-case)',
+        validate: (value) => {
+          if (!value) return '컴포넌트 이름은 필수입니다';
+          if (!/^[a-z0-9]+(?:-[a-z0-9]+)*$/.test(value)) {
+            return '컴포넌트 이름은 kebab-case로 입력해주세요 (예: my-component)';
+          }
+        },
+      })) as string;
+
+      const pascalCaseName = this.kebabToPascal(kebabCaseName);
```

Comment:
```suggestion
      const kebabCaseName = (await text({
        message: '생성할 컴포넌트 이름을 입력해주세요 (kebab-case)',
        placeholder: 'ESC를 눌러 취소할 수 있습니다.',
        validate: (value) => {
          if (!value) return '컴포넌트 이름은 필수입니다';
          if (!/^[a-z0-9]+(?:-[a-z0-9]+)*$/.test(value)) {
            return '컴포넌트 이름은 kebab-case로 입력해주세요 (예: my-component)';
          }
        },
      })) as string;

      if (isCancel(kebabCaseName)) {
        outro('취소되었습니다.');
        return 0;
      }
```

취소 할 수 있도록 안내해주면 UX적으로 괜찮을거같은데 어떻게 생각해 ?

지금은 Control + C 로 종료하면 Type Error 나와서 예외처리가 안된 느낌?

Reply:
오 이거는 너무 좋은 제안인 것 같아